files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,537 @@ class AaaCfg(object):
+
+
+        if (len(intf_keys) < 2):
+            if op == "DELETE":
+                cfgdb_entry['ipv4_dhcp_client'] = "true"
+                cfgdb_entry['ipv6_dhcp_client'] = "true"
+
+                if (('ipv4_dhcp_client' not in appdb_entry) or (appdb_entry['ipv4_dhcp_client'] != cfgdb_entry['ipv4_dhcp_client'])):
+                    self.mgmt_intf_dhclient_set(intf_keys[0], "ipv4_dhcp_client", True)
+                    appdb_entry['ipv4_dhcp_client'] = cfgdb_entry['ipv4_dhcp_client']
+
+                if (('ipv6_dhcp_client' not in appdb_entry) or (appdb_entry['ipv6_dhcp_client'] != cfgdb_entry['ipv6_dhcp_client'])):
+                    self.mgmt_intf_dhclient_set(intf_keys[0], "ipv6_dhcp_client", True)
+                    appdb_entry['ipv6_dhcp_client'] = cfgdb_entry['ipv6_dhcp_client']
+
+                if ('ipv4_default_gwaddr' in appdb_entry):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], appdb_entry['ipv4_default_gwaddr'], "DEL")
+                    del appdb_entry['ipv4_default_gwaddr']
+
+                if ('ipv6_default_gwaddr' in appdb_entry):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], appdb_entry['ipv6_default_gwaddr'], "DEL")
+                    del appdb_entry['ipv6_default_gwaddr']
+
+                self.app_db.set_entry(APP_MGMT_INTF_TABLE, cur_appdb_key, appdb_entry)
+                self.config_db.set_entry(CFG_MGMT_INTF_TABLE, key, cfgdb_entry)
+
+            elif op == "CREATE":
+                syslog.syslog(syslog.LOG_ERR, "handle_mgmt_intf_cfg: operation is CREATE, return after vrf create handling")
+                self.mgmt_vrf_cfg.handle_mgmt_intf_creation(key)
+
+                if ('ipv4_dhcp_client' in cfgdb_entry):
+                    self.mgmt_intf_dhclient_set(intf_keys[0], "ipv4_dhcp_client", cfgdb_entry['ipv4_dhcp_client'] == "true")
+                    appdb_entry['ipv4_dhcp_client'] = cfgdb_entry['ipv4_dhcp_client']
+
+                if ('ipv6_dhcp_client' in cfgdb_entry):
+                    self.mgmt_intf_dhclient_set(intf_keys[0], "ipv6_dhcp_client", cfgdb_entry['ipv6_dhcp_client'] == "true")
+                    appdb_entry['ipv6_dhcp_client'] = cfgdb_entry['ipv6_dhcp_client']
+
+                if ('ipv4_default_gwaddr' in cfgdb_entry):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], cfgdb_entry['ipv4_default_gwaddr'], "ADD")
+                    appdb_entry['ipv4_default_gwaddr'] = cfgdb_entry['ipv4_default_gwaddr']
+
+                if ('ipv6_default_gwaddr' in cfgdb_entry):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], cfgdb_entry['ipv6_default_gwaddr'], "ADD")
+                    appdb_entry['ipv6_default_gwaddr'] = cfgdb_entry['ipv6_default_gwaddr']
+
+                self.app_db.set_entry(APP_MGMT_INTF_TABLE, cur_appdb_key, appdb_entry)
+
+            elif op == "UPDATE":
+                if (('ipv4_dhcp_client' in cfgdb_entry) and ('ipv4_dhcp_client' not in appdb_entry)):
```

Comment:
This section is with repetition and hard to understand. Request to revisit to make it easier to review

Reply:
Done.